hazelcast/src/main/java/com/hazelcast/spi/impl/BasicOperationService.java

```diff
@@ -680,13 +778,21 @@ private void afterRun(Operation op) {
             }
         }
 
-        private void ensureNoPartitionProblems(Operation op) {
+        protected void ensureNoPartitionProblems(Operation op) {
             int partitionId = op.getPartitionId();
+
             if (partitionId < 0) {
                 return;
             }
 
-            InternalPartition internalPartition = nodeEngine.getPartitionService().getPartition(partitionId);
+            if (partitionId != thisPartitionId) {
```

Comment:
Is this a debug assertion? Operations will be already assigned to the correct operation-handler by the service itself. Or can a mis-assignment happen? Otherwise I think you should set a self-explaining exception message:

```
throw new IllegalStateException("bla bla bla...");
```